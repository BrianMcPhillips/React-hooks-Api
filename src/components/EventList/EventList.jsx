import React from 'react';
import PropTypes from 'prop-types';
import Event from '../Event/Event';

const EventList = ({ events }) => {
  const eventStuff = events.map(event => (
    <li key={`${event.year} - ${event.text}`}>
      <Event {...event}/>
    </li>
  ));

  return (
    <ul data-testid="events">
      {eventStuff}
    </ul>
  );
};

EventList.propTypes = {
  events: PropTypes.arrayOf(PropTypes.shape({
    year: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  })).isRequired
};

export default EventList;
