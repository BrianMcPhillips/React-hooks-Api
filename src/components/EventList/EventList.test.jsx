import React from 'react';
import { render, cleanup } from '@testing-library/react';
import EventList from './EventList';

describe('EventList component', () => {
  afterEach(() => cleanup());
  it('renders EventList', () => {
    const { asFragment } = render(
      <EventList 
        events={[
          {
            text:'Best event',
            year:'2022'
          },
          {
            text:'Second best event',
            year:'2024'
          }
        ]}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
