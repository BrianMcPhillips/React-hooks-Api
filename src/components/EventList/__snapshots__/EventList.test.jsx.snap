// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList component renders EventList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h3>
        2022
      </h3>
      <p>
        Best event
      </p>
    </li>
    <li>
      <h3>
        2024
      </h3>
      <p>
        Second best event
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
